test/e2e/app-dir/next-after-app/utils/simulated-invocation.js

```diff
@@ -0,0 +1,184 @@
+import { requestAsyncStorage } from 'next/dist/client/components/request-async-storage.external'
+import { Awaiter } from 'next/dist/server/lib/awaiter'
+import { cliLog } from './log'
+
+// replaced in tests
+const shouldInstallShutdownHook = false
+
+/*
+This module is meant to help simulate a serverless invocation, which will shut down when
```

Comment:
Do we have a deploy test for this? This seems like it'll either break on implementation changes or stops testing the real thing when implementation changes. Not saying this test isn't valueable but there should be some real equivalent.

Reply:
yeah i'm not 100% happy with the testing approach here either...

but the issue is that (AFAIU) in deploy tests, we'll always run on vercel, and we'll have a global `waitUntil` available via `@vercel/request-context`. whereas this PR is trying to fix an issue that occurs when that's *not* available.

i suppose we could try doing a deploy test that deletes/unsets the context and test it that way?

Reply:
It doesn't have to be an automated test. Some repro or fixture or instructions how to test this so that we can check this still works in the future.